impl/client/src/main/doc-js/pentaho/visual/base/spec/IViewType.jsdoc

```diff
@@ -0,0 +1,37 @@
+/*!
+ * Copyright 2010 - 2017 Pentaho Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * The `IViewType` interface represents the set of options used to define and configure a view type.
+ *
+ * @name pentaho.visual.base.spec.IViewType
+ * @interface
+ * @extends pentaho.type.spec.IComplexTypeProto
+ *
+ * @see pentaho.visual.base.View.Type
+ */
+
+/**
+ * Extension properties that a view knows how to handle directly.
+ *
+ * This property is expected to be specified through configuration.
+ *
+ * Custom view classes should document the supported options.
+ *
+ * @name extension
```

Comment:
The `spec.IViewEx` contains instance data, while `extension` is type level metadata. Also the `Ex` in `spec.IViewEx` does not have this meaning — it's just an extended version of the base interface for use in constructors.